### PR TITLE
feat/79: add CategoryContentUpdateDto, mapping, put action, handler, command

### PR DESCRIPTION
**Pull request by NMarteniuk**
…command

dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

File: Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Update/UpdateCategoryContentHandler.cs [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        _mapper = mapper;
+        _logger = logger;
+    }
+
+    public async Task<Result<StreetcodeCategoryContentDto>> Handle(UpdateCategoryContentCommand command, CancellationToken cancellationToken)
+    {
+        CategoryContentUpdateDto request = command.CategoryContentUpdateDto;
+
+        StreetcodeCategoryContent content = _mapper.Map<StreetcodeCategoryContent>(request);
+
```
Comment (yaninavoda) [inline Streetcode/Streetcode.BLL/MediatR/Sources/SourceLinkCategory/Update/UpdateCategoryContentHandler.cs:30]: Here we have to check if content exists in the DB. 
If it does - update it,
if it's not - create it with Text from request and save it to DB

**Review by yaninavoda — DISMISSED** [review]
ok

File: Streetcode/Streetcode.WebApi/Controllers/Source/SourcesController.cs [hunk 2]
```diff
@@ -44,4 +46,10 @@ public async Task<IActionResult> Delete([FromRoute] int id)
     {
         return HandleResult(await Mediator.Send(new DeleteCategoryCommand(id)));
     }
+
+    [HttpPut]
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.WebApi/Controllers/Source/SourcesController.cs:50]: HttpPut пропоную поставити перед HttpDelete

Comment (OlhaPivniuk) [inline Streetcode/Streetcode.WebApi/Controllers/Source/SourcesController.cs:50]: HttpPut пропоную поставити перед HttpDelete

**Review by OlhaPivniuk — COMMENTED** [review]
image

File: Streetcode/Streetcode.BLL/Mapping/Sources/StreetcodeCategoryContentProfile.cs [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
+using Streetcode.BLL.DTO.Sources;
 using Streetcode.DAL.Entities.Sources;
 
 namespace Streetcode.BLL.Mapping.Sources
 {
-    internal class StreetcodeCategoryContentProfile : Profile
+    public class StreetcodeCategoryContentProfile : Profile
     {
-        public StreetcodeCategoryContentProfile()
+        internal StreetcodeCategoryContentProfile()
```
Comment (OlhaPivniuk) [inline Streetcode/Streetcode.BLL/Mapping/Sources/StreetcodeCategoryContentProfile.cs:10]: should be public too

**Review by OlhaPivniuk — APPROVED** [review]

**Review by yaninavoda — APPROVED** [review]

**Pull request merged**